model/modelprocessor/dropunsampled.go

```diff
@@ -51,9 +51,9 @@ func NewDropUnsampled(dropRUM bool, droppedCallback func(int64)) model.BatchProc
 	})
 }
 
-func shouldDropUnsampled(event *model.APMEvent, dropRUM bool) bool {
-	if event.Processor != model.TransactionProcessor || event.Transaction == nil || event.Transaction.Sampled {
+func shouldDropUnsampled(event *modelpb.APMEvent, dropRUM bool) bool {
+	if !event.Processor.IsTransaction() || event.Transaction == nil || event.Transaction.Sampled {
```

Comment:
It would but `IsTransaction` is already using getters under the hood so this should panic even if `event.Processor` is nil.

Is this only for clarity/style or am I missing something ?

Reply:
Not because it is unsafe, but I figured that 2 `||` clauses seem easier to read than 3.

Reply:
Oh sorry, I stopped reading at `Processor` -> `GetProcessor` :facepalm: 

I didn't notice the transaction change, I agree it looks cleaner :+1:

Reply:
Updated!